flask_googlemaps/__init__.py

```diff
@@ -81,6 +86,181 @@ def add_marker(self, lat=None, lng=None, **kwargs):
+        pass the rectangle bounds, but also allows to set each parameter
+        individually if the user wish so.
+
+        Args:
+            north (float): The north latitude bound
+            west (float): The west longitude bound
+            south (float): The south latitude bound
+            east (float): The east longitude bound
+            stroke_color (str): Sets the color of the rectangle border using
+                hexadecimal color notation
+            stroke_opacity (float): Sets the opacity of the rectangle border
+                in percentage. If stroke_opacity = 0, the border is transparent
+            stroke_weight (int): Sets the stroke girth in pixels.
+            fill_color (str): Sets the color of the rectangle fill using
+                hexadecimal color notation
+            fill_opacity (float): Sets the opacity of the rectangle fill
+
+
+        """
+        rectangle = {
+            'stroke_color': stroke_color,
+            'stroke_opacity': stroke_opacity,
+            'stroke_weight': stroke_weight,
+            'fill_color': fill_color,
+            'fill_opacity': fill_opacity,
+            'bounds': {'north': north,
+                       'west': west,
+                       'south': south,
+                       'east': east,
+                       }
+        }
+
+        return rectangle
+
+    def add_rectangle(self,
+                      north=None,
+                      west=None,
+                      south=None,
+                      east=None,
+                      **kwargs):
+        """ Adds a rectangle dict to the Map.rectangles attribute
+
+        The Google Maps API describes a rectangle using the LatLngBounds
+        object, which defines the bounds to be drawn. The bounds use the
+        concept of 2 delimiting points, a northwest and a southeast points,
+        were each coordinate is defined by each parameter.
+
+        It accepts a rectangle dict representation as well.
+
+        Args:
+            north (float): The north latitude
+            west (float): The west longitude
+            south (float): The south latitude
+            east (float): The east longitude
+
+        .. _LatLngBoundsLiteral:
+            https://developers.google.com/maps/documentation/javascript/reference#LatLngBoundsLiteral
+
+        .. _Rectangles:
+            https://developers.google.com/maps/documentation/javascript/shapes#rectangles
+        """
+
+        if north:
+            kwargs['bounds']['north'] = north
+        if west:
+            kwargs['bounds']['west'] = west
+        if south:
+            kwargs['bounds']['south'] = south
+        if east:
+            kwargs['bounds']['east'] = east
+
+        if 'bounds' not in kwargs:
+            raise AttributeError('bounds required to build rectangles')
+
+        if 'bounds' in kwargs \
+                and {'north', 'east', 'south', 'west'} \
+                != kwargs['bounds'].keys():
```

Comment:
that comparison will not work, because dictionaries are unordered.. 

```python
if not all([key in kwargs['bounds'].keys() for key in ('north', 'east', 'south', 'west')])
```